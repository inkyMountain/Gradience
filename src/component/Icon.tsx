import React from 'react';
import '../static/icons/index.js';


interface IconProps {
  name?: string;
}

const Icon = (props: IconProps) => {
  return <div className='component-icon'>
    <svg>
      <use xlinkHref={`#${props.name}`}/>
    </svg>
  </div>;
};

export {Icon};
