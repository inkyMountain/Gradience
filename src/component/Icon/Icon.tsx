import React from 'react';
import '../../static/icons/index.js';
import './icon.scss';

interface IconProps extends React.HTMLAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> =
  (props) => {
    const {name, className, ...restProps} = props;
    return <svg className={`component-icon ${className || ''}`} {...restProps}>
      <use xlinkHref={`#${name}`}/>
    </svg>;
  };

export default Icon;
