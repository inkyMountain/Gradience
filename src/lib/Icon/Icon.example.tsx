import React, {Fragment} from 'react';
import Icon from './Icon';

const IconExample = () => {
  return (
    <Fragment>
      <Icon name='wechat'/>
      <Icon name='alipay'/>
      <Icon name='right'/>
    </Fragment>
  );
};

export default IconExample;
