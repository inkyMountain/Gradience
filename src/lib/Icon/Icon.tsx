import React from 'react';
import '../static/icons/index.js';
import './Icon.scss';
import classes from '../utils/classes';

interface IconProps extends React.HTMLAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> = props => {
  const {name, className, ...restProps} = props;
  return <svg className={classes('component-icon', className)} {...restProps}>
    <use xlinkHref={`#${name}`}/>
  </svg>;
};

export default Icon;
