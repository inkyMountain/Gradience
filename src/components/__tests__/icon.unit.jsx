import Icon from '../Icon/Icon';
import * as renderer from 'react-test-renderer';
import React from 'react';
import {mount} from 'enzyme';

describe('components PageIcon', () => {
  it('PageIcon UI is correct', () => {
    const iconJson = renderer.create(<Icon name={'wechat'}/>).toJSON();
    expect(iconJson).toMatchSnapshot();
  });

  it('can be clicked', function () {
    const fn = jest.fn();
    const icon = mount(<Icon name={'wechat'} onClick={fn}/>);
    icon.simulate('click');
    expect(fn).toBeCalled();
  });
});
