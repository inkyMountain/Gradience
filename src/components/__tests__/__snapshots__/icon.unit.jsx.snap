// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component icon icon UI is correct 1`] = `
<svg
  className="component-icon "
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
